lib/WasmReader/WasmByteCodeGenerator.cpp

```diff
@@ -1600,7 +1600,7 @@ void WasmBytecodeGenerator::YieldToBlock(BlockInfo blockInfo, EmitInfo expr)
         if (!IsUnreachable())
         {
             blockInfo.yieldInfo->didYield = true;
-            m_writer->AsmReg2(GetLoadOp(expr.type), yieldInfo.location, expr.location);
+            m_writer->AsmReg2(GetReturnOp(expr.type), yieldInfo.location, expr.location);
```

Comment:
I considered making a `Yield_***` opcode, but I was able to get the same result with `Return_***` might need to revisit when we'll do multi-return